orttraining/orttraining/core/session/training_session.cc

```diff
@@ -48,6 +48,11 @@ Status SetupOptimizerParams(
   ORT_RETURN_IF_NOT(config.optimizer_config.has_value());
   const auto& optimizer_config = config.optimizer_config.value();
 
+  TrainingSession::OptimizerState init_optimizer_states;
+  if (config.init_optimizer_states.has_value()){
```

Comment:
>has_value() [](start = 35, length = 11)

just FYI, we have a more fully featured optional now with operator bool :)
so we could say `if (config.init_optimizer_states)`